io.github.andrewmkiss.Demeter.yml

```diff
@@ -0,0 +1,190 @@
+      - '/bin/perlthanks'
+      - '/bin/piconv'
+      - '/bin/pl2pm'
+      - '/bin/pod*'
+      - '/bin/prove'
+      - '/bin/ptar*'
+      - '/bin/shasum'
+      - '/bin/splain'
+      - '/bin/xsubpp'
+      - '/bin/zipdetails'
+      - '/include'
+      - '/man'
+      - '*.pod'
+
+  ## Using the dependencies from flatpak-cpan-generator
+  # Installs the modules generated by flatpak-cpan-generator.
+  # Note that *any* Makefile.PL-style modules MUST be installed in this one step,
+  # as once perllocal.pod is written in one module, it cannot be modified by others.
+  - name: perl-libs
+    disabled: false
+    buildsystem: simple
+    build-commands:
+      - 'perl-libs/install.sh'
+    # Same as with the Perl module, we need to restore write permission.
+    # However, -f is now passed to avoid errors from trying to touch files from the
+    # above module that are now marked as read-only.
+    post-install:
+      - 'chmod -Rf u+w /app/lib/perl5/site_perl'
+      - 'chmod u+w /app/bin/pdl*'
+    sources:
+      - perl-libs.yml
+    # This step should be customized based on the CPAN packages you're using.
+    cleanup:
+      - '/bin'
+      - '/man'
+
+  ## Demeter
+  - name: demeter
+    disabled: false
+    buildsystem: simple
+    build-options:
+      env:
+        - PERL5LIB=/run/build/demeter:/app/lib/perl5/site_perl
+        - PERLLIB=/app/lib/perl5/site_perl/5.36.1
+    build-commands:
+      - perl Build.PL
+      - sed -i "1653,+3s/25/35/" lib/Demeter/UI/Athena.pm
+      - ./Build
+      - ./Build install
+      - install -Dm 644 metainfo.xml /app/share/metainfo/io.github.andrewmkiss.Demeter.metainfo.xml
+      - install -Dm 644 /app/lib/perl5/site_perl/5.36.1/x86_64-linux/Demeter/share/Demeter_icon.png /app/share/icons/hicolor/48x48/apps/io.github.andrewmkiss.Demeter.png
+      - install -Dm 644 /app/lib/perl5/site_perl/5.36.1/x86_64-linux/Demeter/UI/Athena/share/athena_icon.png /app/share/icons/hicolor/48x48/apps/io.github.andrewmkiss.Demeter.Athena.png
+      - install -Dm 644 /app/lib/perl5/site_perl/5.36.1/x86_64-linux/Demeter/UI/Artemis/share/artemis_icon.png /app/share/icons/hicolor/48x48/apps/io.github.andrewmkiss.Demeter.Artemis.png
+      - ffmpeg -i Demeter.png -s 128x128 -frames:v 1 Demeter_128x128.png
+      - install -Dm 644 Demeter.png /app/share/app-info/icons/flatpak/512x512/io.github.andrewmkiss.Demeter.png
+      - install -Dm 644 Demeter_128x128.png /app/share/app-info/icons/flatpak/128x128/io.github.andrewmkiss.Demeter.png
+      - install -Dm 644 -t /app/share/applications io.github.andrewmkiss.Demeter.Athena.desktop
+      - install -Dm 644 -t /app/share/applications io.github.andrewmkiss.Demeter.Artemis.desktop
+      - install -Dm 644 -t /app/share/applications io.github.andrewmkiss.Demeter.Hephaestus.desktop
+    post-install:
+      - 'chmod u+w /app/lib/perl5/site_perl/5.36.1/x86_64-linux/auto/Ifeffit/Ifeffit.so'
+    sources:
+      - type: git
+        url: https://github.com/bruceravel/demeter
```

Comment:
why does this not match the app-id ?

Reply:
the app-id is should be related to the application being packaged (aka under the control of upstream developers). So the one you chose doesn't meet the requirements.

Reply:
Okay, this comment seems important.
If I understand what you are saying, the proper name for this application would really be something like `io.github.bruceravel.Demeter` since he is the developer and owner of the main package. It is not allowed to rename the package, as I am, in my own repo where I would be in control of the "flatpak", but not the upstream source code. 

I did speak with the upstream developer and received a verbal "blessing" but I'm guessing you probably want something more official than that.

Given the complexity of this software with perl dependencies etc., I thought this would be a good candidate for a flatpak, and then wider distribution on flathub for the scientific community.

Reply:
It should be using the upstream namespace.